cmd/envoy/main.go

```diff
@@ -534,93 +532,114 @@ func daybreakImport(c *cli.Context) (err error) {
 		return cli.Exit(err, 1)
 	}
 
-	// db context
+	// Make sure we have a daybreak database
+	ddb, ok := db.(store.DaybreakStore)
+	if !ok {
+		return cli.Exit("configured database does not support daybreak operations", 2)
+	}
+
+	// Create outer context for database interactions and source info.
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	// Load counterparty IDs currently in the DB and put into a map
-	var srcInfo []*models.CounterpartySourceInfo
-	if srcInfo, err = db.ListCounterpartySourceInfo(ctx, enum.SourceDaybreak); err != nil {
+	// NOTE: this operation happens in its own transaction, if two daybreak imports
+	// happen concurrently, then this map would not accurately reflect the state of the
+	// database, so its important to ensure only a single import runs at at time.
+	var srcMap map[string]*models.CounterpartySourceInfo
+	if srcMap, err = ddb.ListDaybreak(ctx); err != nil {
 		return cli.Exit(err, 1)
 	}
 
-	srcMap := make(map[string]*models.CounterpartySourceInfo)
-	for _, cSrc := range srcInfo {
-		if cSrc.DirectoryID.Valid {
-			srcMap[cSrc.DirectoryID.String] = cSrc
-		} else {
-			log.Warn().Msg(fmt.Sprintf("CounterpartySourceInfo was missing a DirectoryId (ID: %s)", cSrc.ID))
-		}
-	}
+	// Begin import; all counterparties and associated contacts that can be created will
+	// be; otherwise the entire counterparty will be skipped.
+	log.Info().Msgf("starting to import %d Daybreak Counterparties...", len(cpartyImports))
 
-	log.Info().Msg(fmt.Sprintf("Starting to import %d Daybreak Counterparties...", len(cpartyImports)))
+	var (
+		updated int
+		created int
+	)
```

Comment:
I broke up `importCnt` into two counters to make it easier to see what was happening.